src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketPostClient.java

```diff
@@ -0,0 +1,26 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.client.exception.*;
+
+/**
+ * Basic Bitbucket post client. A client is normally threadsafe and can be used multiple times.
+ *
+ * @param <ResponseType> The data returned from the POST request
+ * @param <RequestType> The data to be sent in the POST request
+ */
+public interface BitbucketPostClient<ResponseType, RequestType> {
+
+    /**
+     * Make the call out to Bitbucket and read the response.
```

Comment:
These are basically the same as those in the in the `HttpRequestExecutor.java` post method.